src/ert/_c_wrappers/enkf/lark_parser.py

```diff
@@ -0,0 +1,419 @@
+        current = current.value
+
+    return current
+
+
+class _TreeToDictTransformer:
+    def __init__(
+        self,
+        config_file: str,
+        config_dir: str,
+        config_file_base: str,
+        add_invalid: bool = False,
+    ):
+        self.defines = []
+        self.data_kws = []
+        self.keywords = []
+        self.config_dict = None
+        self.errors = []
+        self.add_invalid = add_invalid
+        self.config_dir = config_dir
+        self.config_file_base = config_file_base
+        self.config_file = config_file
+
+    def __call__(self, tree: Tree, site_config: Optional[Dict[str, Any]] = None):
+        schema: Mapping[str, SchemaItem] = (
+            init_user_config() if site_config is not None else init_site_config()
+        )
+
+        self.defines.append(["<CONFIG_PATH>", self.config_dir])
+        self.defines.append(["<CONFIG_FILE_BASE>", self.config_file_base])
+        self.defines.append(["<DATE>", datetime.date.today().isoformat()])
+        self.schema = schema
+        self.config_dict = {} if not site_config else site_config
+        for node in tree.children:
+            if isinstance(node, Tree) and node.data == "instruction":
+                if isinstance(node.children[0], Token):
+                    pass  # newline or comment
+                elif node.children[0].data == "define":
+                    self.define(node.children[0])
+                elif node.children[0].data == "data_kw":
+                    self.data_kw(node.children[0])
+                elif node.children[0].data == "keyword":
+                    self.keyword(node.children[0])
+                elif node.children[0].data == "include":
+                    self.include(node.children[0])
+
+        def check_valid(val, item: SchemaItem, index: int):
+            if index in item.indexed_selection_set:
+                if val not in item.indexed_selection_set[index]:
+                    raise ConfigValidationError(
+                        f"{item.kw!r} argument {index!r} must be one of"
+                        f" {item.indexed_selection_set[index]!r}"
+                    )
+
+        def convert(val, item: SchemaItem, index: int):
+            check_valid(val, item, index)
+            if not len(item.type_map) > index:
+                return val
+            val_type = item.type_map[index]
+            if val_type is None:
+                return val
+            if val_type == SchemaType.CONFIG_INT:
+                try:
+                    return int(val)
+                except ValueError:
+                    raise ConfigValidationError(
+                        f"{item.kw!r} must have an integer value"
+                        f" as argument {index + 1!r}"
+                    ) from None
+            if val_type == SchemaType.CONFIG_FLOAT:
+                try:
+                    return float(val)
+                except ValueError:
+                    raise ConfigValidationError(
```

Comment:
This failure condition is not tested, ie. there is no test checking that you get a reasonable error message using a non-float where floats are expected.